src/aggregation/metric/stats.rs

```diff
@@ -232,22 +233,23 @@ impl SegmentAggregationCollector for SegmentStatsCollector {
         doc: crate::DocId,
         agg_with_accessor: &AggregationsWithAccessor,
     ) -> crate::Result<()> {
-        let field = &agg_with_accessor.metrics.values[self.accessor_idx].accessor;
+        let field = unsafe {
+            &agg_with_accessor
+                .metrics
+                .values
+                .get_unchecked(self.accessor_idx)
```

Comment:
is the unsafe making a difference?

Reply:
Yes, but the benchmarks are a little flaky. I'll remove the unsafe for now